content/en/docs/concepts/architecture/nodes.md

```diff
@@ -513,6 +513,30 @@ During a non-graceful shutdown, Pods are terminated in the two phases:
   recovered since the user was the one who originally added the taint.
 {{< /note >}}
 
+### Forced storage detach on timeout {#storage-force-detach-on-timeout}
+
+In any situation where a pod deletion has not succeeded for 6 minutes, kubernetes will
+force detach volumes being unmounted if the node is unhealthy at that instant. Any
+workload still running on the node that uses a force-detached volume will cause a
+violation of the
+[CSI specification](https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerunpublishvolume),
+which states that ControllerUnpublishVolume "MUST be called after all
+NodeUnstageVolume and NodeUnpublishVolume on the volume are called and succeed".
+In such circumstances, volumes on the node in question might encounter data corruption.
+
+With the advent of the "Non-graceful node shutdown" feature, users might opt to disable
+this behavior.
+
+Force storage detach on timeout can be disabled by setting the `disable-force-detach-on-timeout`
+config field in `kube-controller-manager`. Disabling the force detach on timeout feature means
+that a volume that is hosted on a node that is unhealthy for more than 6 minutes will not have
+its associated
+[VolumeAttachment](/docs/reference/kubernetes-api/config-and-storage-resources/volume-attachment-v1/)
+deleted.
+
+After this setting has been applied, unhealthy pods still attached to a volumes must be recovered
+via the [Non-Graceful Node Shutdown](#non-graceful-node-shutdown) procedure mentioned above.
```

Comment:
Can you add a note that the user needs to use caution when using the "non graceful node shutdown" feature?  See the "Note" section of https://kubernetes.io/docs/concepts/architecture/nodes/#non-graceful-node-shutdown.  Otherwise, they could still encounter data corruption.

Reply:
Thanks; I'll try to add something to that effect.